src/SDKs/SqlManagement/Sql.Tests/ElasticPoolCrudScenarioTests.cs

```diff
@@ -98,6 +102,71 @@ public void TestUpdateElasticPoolWithUpdateAndListActivity()
             };
         }
 
+        [Fact]
+        public async Task TestCancelUpdateElasticPoolOperation()
+        {
+            /* *
+             * In this test we only test the cancel operation on resize pool from Premium to Premium
+             *    since currently we only support Cancel pool resize operation on Premium <-> Premium
+             * */
+            string testPrefix = "sqlelasticpoollistcanceloperation-";
+            using (SqlManagementTestContext context = new SqlManagementTestContext(this))
+            {
+                ResourceGroup resourceGroup = context.CreateResourceGroup("North Europe");
+                Server server = context.CreateServer(resourceGroup, "northeurope");
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+                Dictionary<string, string> tags = new Dictionary<string, string>()
+                {
+                    { "tagKey1", "TagValue1"}
+                };
+
+                // Create a premium elastic pool with required parameters
+                string epName = SqlManagementTestUtilities.GenerateName();
+                var epInput = new ElasticPool()
+                {
+                    Location = server.Location,
+                    Edition = DatabaseEdition.Premium,
+                    Tags = tags,
+                    Dtu = 125
+                };
+                var elasticPool = sqlClient.ElasticPools.CreateOrUpdate(resourceGroup.Name, server.Name, epName, epInput);
+                SqlManagementTestUtilities.ValidateElasticPool(epInput, elasticPool, epName);
+                Assert.NotNull(elasticPool);
+
+                // Update elastic pool to Premium with 250 DTU
+                var epUpdateReponse = sqlClient.ElasticPools.BeginCreateOrUpdateWithHttpMessagesAsync(resourceGroup.Name, server.Name, epName, new ElasticPool()
+                {
+                    Location = server.Location,
+                    Edition = DatabaseEdition.Premium,
+                    Dtu = 250,
+                    Tags = tags
+                });
+                Thread.Sleep(TimeSpan.FromSeconds(5));
```

Comment:
We should only sleep when in record mode. Take a look at other tests e.g. Restore tests

```
                    // No need to sleep if we are playing back the recording.
                    if (HttpMockServer.Mode == HttpRecorderMode.Record)
                    {
                        Thread.Sleep(TimeSpan.FromSeconds(5));
                    }
```

Reply:
btw how reliable is this test? Does it generally pass every time?

Reply:
yes, when I run the test for multiple time, it passed every time.

Reply:
@jaredmoo update the test in the new commit of the PR: only sleep when in Record mode